cinspect/dependence.py

```diff
@@ -139,7 +140,12 @@ def construct_grid(
     grid_counts = None
     grid = None
 
-    if isinstance(grid_values, Union[List, Tuple, np.ndarray]):
+    def is_list_tuple_or_array(
+        x,
+    ):  # For Python >3.10:  -> TypeGuard[Union[List, Tuple, np.ndarray]]:
+        return isinstance(x, List) or isinstance(x, Tuple) or isinstance(x, np.ndarray)
```

Comment:
you could also do this: `isinstance(x, (List, Tuple, np.ndarray))`

See: https://www.w3schools.com/python/ref_func_isinstance.asp

Then you may not have to make this into a function

Reply:
Oh well that's much nicer (and presumably gets `mypy` type narrowing support for free as well)